File: src/main/java/com/example/appforauslenderamt/entity/ResidencePermitValidity.java
package com.example.appforauslenderamt.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Value;

@Value
@Builder
@Getter
@AllArgsConstructor
public class ResidencePermitValidity {

    @JsonProperty("days")
    Integer days;
    @JsonProperty("months")
    Integer months;
    @JsonProperty("years")
    Integer years;

}

File: Frontend/src/app/Models/BackendModels/ResidencePermitValidity.ts
import { Expose } from "class-transformer"

export class ResidencePermitValidity {
    @Expose()
    days: number
    @Expose()
    months: number
    @Expose()
    years: number

    constructor(days: number, months: number, years: number) {
        this.days = days
        this.months = months
        this.years = years
    }

}
